keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -77,17 +78,16 @@ class ArtifactListener(
 
     if (repository.artifactVersions(artifact).isEmpty()) {
       val artifactSupplier = artifactSuppliers.supporting(artifact.type)
-      val latestArtifact = runBlocking {
+
+      val latestVersion = runBlocking {
         log.debug("Retrieving latest version of registered artifact {}", artifact)
         artifactSupplier.getLatestArtifact(artifact.deliveryConfig, artifact)
       }
-      val latestVersion = latestArtifact?.let { artifactSupplier.getFullVersionString(it) }
+
       if (latestVersion != null) {
-        var status = artifactSupplier.getReleaseStatus(latestArtifact)
-        log.debug("Storing latest version {} (status={}) for registered artifact {}", latestVersion, status, artifact)
-        repository.storeArtifact(artifact.name, artifact.type, latestVersion, status)
-        // send an event when the artifact is stored, so we can fetch its metadata
-        publisher.publishEvent(ArtifactSaved(latestArtifact, status))
+        log.debug("Storing latest version {} (status={}) for registered artifact {}", latestVersion.version, latestVersion.status, artifact)
+        val enrichedArtifact = artifactSupplier.addMetadata(latestVersion.normalized())
+        repository.storeArtifactVersion(enrichedArtifact)
```

Comment:
Instead of doing one write to `ARTIFACT_VERSIONS` to store the basics here, and another as a reaction to the `ArtifactSaved` event just below, consolidate in a single call to `storeArtifactVersion()`.